kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -469,30 +571,34 @@ define([
             container = parent.appendChild(document.createElement('div'));
             ui = UI.make({ node: container });
 
-            const events = Events.make();
-
             container.innerHTML = layout();
-            events.attachEvents(container);
 
             if (config.initialValue !== undefined) {
                 // note this might come from a different workspace...
                 model.value = config.initialValue;
             }
+            if (config.initialDisplayValue !== undefined) {
+                model.displayValue = config.initialDisplayValue;
+            }
 
             return render().then(() => {
                 channel.on('reset-to-defaults', () => {
                     resetModelValue();
                 });
                 channel.on('update', (message) => {
-                    setModelValue(message.value);
+                    setModelValue(message.value, message.displayValue);
                 });
-                setControlValue(getModelValue());
+                updateControlValue();
                 return autoValidate();
             });
         }
 
         function stop() {
             return Promise.try(() => {
+                const control = ui.getElement('input-container.input');
+                if (control) {
+                    $(control).select2('destroy').html('');
+                }
```

Comment:
Select2 puts all kinds of things in all kinds of places in the DOM. We need to specifically call `destroy` to make sure it all goes away.